crates/runtime/src/accelerated_table.rs

```diff
@@ -400,64 +427,91 @@ impl AcceleratedTable {
     }
 }
 
-fn get_expr_time_format(
-    field: Option<(usize, &arrow::datatypes::Field)>,
-    time_format: &Option<TimeFormat>,
-) -> Option<ExprTimeFormat> {
-    let expr_time_format = match field {
-        Some(field) => match field.1.data_type() {
-            DataType::Int8
-            | DataType::Int16
-            | DataType::Int32
-            | DataType::Int64
-            | DataType::UInt8
-            | DataType::UInt16
-            | DataType::UInt32
-            | DataType::UInt64
-            | DataType::Float16
-            | DataType::Float32
-            | DataType::Float64 => {
-                let mut scale = 1;
-                if let Some(time_format) = time_format.clone() {
-                    if time_format == TimeFormat::UnixMillis {
-                        scale = 1000;
+#[derive(Clone, Debug)]
+struct TimestampFilterConvert {
+    time_column: String,
+    time_format: ExprTimeFormat,
+}
+
+#[allow(clippy::needless_pass_by_value)]
+impl TimestampFilterConvert {
```

Comment:
This seems unrelated to the accelerated_table - let's put this in a separate module